.gitlab-ci.yml

```diff
@@ -484,6 +484,31 @@ stop-review-app:
 include:
   - template: Jobs/SAST.gitlab-ci.yml
   - template: Jobs/Dependency-Scanning.gitlab-ci.yml
+  - template: Security/Secret-Detection.gitlab-ci.yml
+
+secret_detection:
+  allow_failure: false
+  variables:
+      SECRET_DETECTION_LOG_OPTIONS: origin/${CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_NAME}..HEAD
+      SECRET_DETECTION_REPORT_FILE: "gl-secret-detection-report.json"
+  rules:
+    - if: $SECRET_DETECTION_DISABLED
+      when: never
+    - if: '$CI_COMMIT_BRANCH || $CI_COMMIT_TAG'
+  before_script:
+    - apk add --no-cache jq
+  script:
+    - /analyzer run
+    # check if '{ "vulnerabilities": [], ..' is empty in the report file if it exists
+    - |
+      if [ -f "$SECRET_DETECTION_REPORT_FILE" ]; then
+        if [ "$(jq ".vulnerabilities | length" $SECRET_DETECTION_REPORT_FILE)" -gt 0 ]; then
+          echo "Vulnerabilities detected. Please analyze the artifact $SECRET_DETECTION_REPORT_FILE produced by the 'secret-detection' job."
+          exit 80
```

Comment:
Q: is the `80` relevant or can it be any non-zero exit code?

Reply:
I don't know, @mitchellhenke did this part.

Reply:
This was actually copied from [this forum post](https://forum.gitlab.com/t/fail-the-pipeline-if-secret-detection-finds-vulnerabilities/56950/4). As far as I can tell, Gitlab fails on any non-zero exit code, like you said. Would you like me to change it, or add a comment?